db/snaptype/type.go

```diff
@@ -167,7 +166,28 @@ func (i Index) HasFile(info FileInfo, logger log.Logger) bool {
 
 	defer segment.Close()
 
-	idx, err := recsplit.OpenIndex(filepath.Join(dir, fName))
+	fNameMask := IdxFileMask(info.From, info.To, i.Name)
+	fPath, fileVer, ok, err := version.FindFilesWithVersionsByPattern(filepath.Join(dir, fNameMask))
+	if err != nil {
+		return false
+	}
+
+	if !ok {
+		_, fName := filepath.Split(fPath)
+		logger.Debug("[ind] HasFile: file does not exists", "f", fName)
+		return false
+	}
+
+	if !fileVer.Eq(i.Version.Current) {
```

Comment:
this line smells - because Erigon with current verison `1.2.3` must be able to open files of version `1.2.4`